pgcli/main.py

```diff
@@ -1385,7 +1387,21 @@ def cli(
                 fg="red",
             )
             exit(1)
-        pgcli.connect_uri(dsn_config)
+        kwargs = {
+            "host": host,
+            "port": port,
+            "username_opt": username_opt,
+            "dbname_opt": dbname_opt,
+        }
+        remap = {"username_opt": "user", "dbname_opt": "dbname"}
+        kwargs = {
+            remap.get(k, k): v
+            for k, v in kwargs.items()
+            if ctx.get_parameter_source(k) is click.core.ParameterSource.COMMANDLINE
+        }
```

Comment:
can you add testcases for this part? thanks

Reply:
Specifically, on a test machine, we have many instances of the PostgreSQL service running on random ports (when creating an instance, our administration system randomly selects an available port for that instance), so I can't easily connect to each instance by setting multiple DSN. I hope that when using a DSN whose IP address is set to our test machine to connect, I can specify the port obtained from our administration system with the `-p` option. But currently pgcli ignore my `-p` option when I use a DSN to connect, even if this DSN does not explicitly specify a port (if no port is specified, pgcli will use the default port 5432 to connect)

Reply:
Well, I add a script to test this part, I mock `PGCli.connect_uri` to verify that whether using `-D dsn` along with `-h` `-p` `-u` `-d` `-W` can override the connection information of `dsn`. But I'm not familiar with pytest, so let me know if there are any problems in my test script